### ISSUE TITLE
Failed prop type: Invalid prop `appElement` supplied to `Modal`.

### ISSUE THREAD

**Issue by moroshko** [reactjs/react-modal#871]
Currently, `appElement` is expected to be an instance of `window.HTMLElement`.

But, when the `Modal` is rendered inside an `iframe`, the iframe's `window.HTMLElement` is a different instance causing the following warning:

> Failed prop type: Invalid prop `appElement` supplied to `Modal`.

See: [Codesandbox](https://codesandbox.io/s/react-modal-in-iframe-forked-0iykb)

What's the best to workaround this?

**Comment by diasbruno** [comment]
Hi @moroshko, this is an interest case. Probably there is no workaround, because if we are dealing with 2 DOM/frame instances, I believe every property that expects an elements will fail.

Probably we need another strategy to validate these properties.

Any Idea?

**Comment by moroshko** [comment]
Is setting and removing attributes the only thing we do with the `appElement`?
If so, what if we had a PropTypes function that checked whether `appElement` has a `setAttribute` (and maybe `removeAttribute`) methods?

**Comment by diasbruno** [comment]
Perhaps `node type` or something...

**Comment by moroshko** [comment]
Do you mean `PropTypes.node`? It doesn't feel right, since it can be a number, for example.

**Comment by diasbruno** [comment]
Something like...[facebook/prop-types#proptypescheckproptypes](https://github.com/facebook/prop-types#proptypescheckproptypes)

```js
const myPropTypes = {
  nodeType: PropTypes.number,
  setAttribute: PropTypes.func,
  removeAttribute: PropTypes.func,
  ....
};

const props = HTMLElement;

PropTypes.checkPropTypes(myPropTypes, props, 'prop', 'MyComponent');
```

**Comment by diasbruno** [comment]
@moroshko I've pushed a branch here `fix/appElement-property-detection`...can you check if it fixes the problem, please?

**Comment by moroshko** [comment]
How could I try this branch in my app? It doesn't contain the `lib` folder which `main` in `package.json` points to. I get:
> Can't resolve 'react-modal'

Could you publish a version to npm?

**Comment by moroshko** [comment]
@diasbruno Any chance to publish this fix to npm (even as a beta version)?

**Comment by diasbruno** [comment]
@moroshko I fixed the reference for this test...you can use the same branch.

**Comment by moroshko** [comment]
@diasbruno I'm afraid it's not easy to test it the way you propose. I'm getting various errors like: 
```
SyntaxError: Unexpected token
``` 
and 
```
Support for the experimental syntax 'classProperties' isn't currently enabled
```
I think we need to compile it. Can we please publish a beta version to npm for testing? This way we are testing the real thing.

**Comment by diasbruno** [comment]
Yeah...the project will need to have everything to react-modal uses to compile...I'll upload a precompiled version in this branch.

**Comment by diasbruno** [comment]
Added the precompiled version. Hope now we can test this. 🤞

**Comment by moroshko** [comment]
The original warning has gone, but now we get:
> Warning: Invalid argument supplied to oneOfType. Expected an array of check functions, but received an object at index 0.

**Comment by moroshko** [comment]
@diasbruno Any thoughts how to proceed from here?

**Comment by andrewpye** [comment]
Has there been any update on this? I've just started running into the same issue too. Thanks!

**Comment by doougui** [comment]
Updates?
